components/server/src/workspace/garbage-collector.ts

```diff
@@ -41,16 +43,25 @@ export class WorkspaceGarbageCollector {
     }
 
     public async garbageCollectWorkspacesIfLeader() {
-        if (await this.leaderQuorum.areWeLeader()) {
-            log.info("wsgc: we're leading the quorum. Collecting old workspaces");
-            this.softDeleteOldWorkspaces().catch((err) => log.error("wsgc: error during soft-deletion", err));
-            this.deleteWorkspaceContentAfterRetentionPeriod().catch((err) =>
-                log.error("wsgc: error during content deletion", err),
-            );
-            this.purgeWorkspacesAfterPurgeRetentionPeriod().catch((err) =>
-                log.error("wsgc: error during hard deletion of workspaces", err),
-            );
-            this.deleteOldPrebuilds().catch((err) => log.error("wsgc: error during prebuild deletion", err));
+        try {
+            await this.mutex.client().using(["workspace-gc"], 30 * 1000, async (signal) => {
```

Comment:
What's the reason for having a 30 second timeout? Isn't this quite long so that two concurrent runs could both succeed when the first run takes less than 30 sec?

Reply:
Makes sense. I didn't look up the API definition and assumed the argument would do something else.